### PR TITLE
Low-rank DMD: Optimal solution in poylnomial time

### PR DESCRIPTION
**Pull request by loiseaujc**
Implementation of the optimal solution by Héas & Herzet.

- New DMD object : optdmd --> name could be changed obviously.
- Added a self.predict(X) function to both DMD and optDMD, following scikit-learn API to make predictions.
- Added tutorial 8 --> Work in Progress.

### CODE REVIEW

File: pydmd/optdmd.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# Author : Jean-Christophe Loiseau <jean-christophe.loiseau@ensam.eu>
```
Comment (mtezzele) [inline pydmd/optdmd.py]: Please remove these lines. The history of the file already shows the single contributions. Add instead the docstring of the module please.

File: pydmd/optdmd.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import numpy as np
+from scipy.linalg import eig, eigvals
+from scipy.linalg import svdvals
+from scipy.linalg import pinv2
+
+
+def pinv(x): return pinv2(x, rcond=10*np.finfo(float).eps)
+
+
+class optDMD(DMDBase):
```
Comment (mtezzele) [inline pydmd/optdmd.py]: Change the name with OptDMD (camel convention). Remember also to change it in the init and in the tutorial, please.

File: pydmd/dmd.py [hunk 3]
```diff
@@ -1,6 +1,14 @@
 """
 Derived module from dmdbase.py for classic dmd.
 """
+# --> Import standard python packages
+import numpy as np
+from scipy.linalg import eig, eigvals
```
Comment (mtezzele) [inline pydmd/dmd.py]: condense these 2 lines in one and remove the unused eigvals

File: pydmd/optdmd.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+    """
+    Dynamic Mode Decomposition
+
+    This class implements the closed-form solution to the DMD minimization
+    problem. It relies on the optimal solution given by Héas & Herzet [1].
```
Comment (mtezzele) [inline pydmd/optdmd.py]: Remove the non-ASCII characters please, otherwise the tests fail on travis.

File: pydmd/optdmd.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        to reach the 'energy' specified by `svd_rank`; if -1, the method does
+        not compute truncation.
+
+    Attributes
+    -----------
+    ???
+
+    References
+    ----------
+    [1] P. Héas & C. Herzet. Low-rank dynamic mode decomposition: optimal
```
Comment (mtezzele) [inline pydmd/optdmd.py]: Same as above

**Comment by loiseaujc** [comment]
Hej,

Thanks a lot for the positive feedback.
I'll do all of the minor changes and add the tests either by the end of this week or early next week.

As for tutorial 1 and 2, I indeed did not change anything, simply re-run them. I guess I didn't pay attention that they had been gitted. I'll change this as well.

++
JC

**Comment by loiseaujc** [comment]
Oups, I forgot to add the test. I will resubmit the PR tomorrow with the tests added.

**Pull request merged**